### PR TITLE
feat(VEG-1285): refactor app example by using zendesk react garden components

### PR DESCRIPTION
**Pull request by cl8au**
🤞 

## Description
- chore: upgrade node to v18 with v5 webpack
- feat: uplift existing code by using react garden design
- doc: update README to contains React component and other updates
- doc: remove IE support in browserslist
- ci: add github actions for ci
- ci: enable lint in github action

Zendesk CSS components are out of support. We'd like to uplift this scaffold to rely on Zendesk React components.

## Tasks
- [x] Write tests

## References
- JIRA: https://zendesk.atlassian.net/browse/VEG-1285

## Screenshots (if needed)
Screen Shot 2022-12-21 at 16 24 36

Screen Shot 2022-12-21 at 16 24 51

## Risks
* [Medium]: Refactor the example app by using Zendesk React component

### CODE REVIEW

**Review by Frederick888 — APPROVED** [review]
Amazing! Deserves at least a minor version bump I think 😆

File: webpack/translations-loader.js [hunk 1]
```diff
@@ -22,6 +22,7 @@
  *   app.steps.click: 'this button'
  * }
  */
+/* eslint-disable array-callback-return */
```
Comment (cl8au) [inline webpack/translations-loader.js:25]: note: I think it is a little bit risky to refactor this function. The issue is the function using map as loop to go through all the key/value pair.. Just ignore this function only from the lint error array-callback-return in order to add lint as part of the GA

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "app_scaffold",
-  "version": "1.0.1",
+  "version": "1.0.2",
```
Comment (JeanMarcGoepfert) [inline package.json:3]: Should probably be a major version bump, considering we're dropping support for older node/ruby versions?

**Review by anushkafka — APPROVED** [review]
LGTM 👍

File: package.json (hunk 2 above)
Comment (cl8au) [reply package.json:3]: Sure, I can update it to be 2.0.0. :)

Comment (cl8au) [reply package.json]: Addressed :)

**Review by JeanMarcGoepfert — APPROVED** [review]

**Pull request merged**